libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/FileInterceptor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.javaagent;
+
+import org.opensearch.javaagent.bootstrap.AgentPolicy;
+
+import java.io.FilePermission;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.Policy;
+import java.security.ProtectionDomain;
+import java.util.List;
+
+import net.bytebuddy.asm.Advice;
+
+/**
+ * FileInterceptor
+ */
+public class FileInterceptor {
+    /**
+     * FileInterceptor
+     */
+    public FileInterceptor() {}
+
+    /**
+     * Intercepts file operations
+     *
+     * @param args   arguments
+     * @param method method
+     * @throws Exception exceptions
+     */
+    @Advice.OnMethodEnter
+    @SuppressWarnings("removal")
+    @SuppressForbidden(reason = "Using FilePermissions directly is ok")
+    public static void intercept(@Advice.AllArguments Object[] args, @Advice.Origin Method method) throws Exception {
+        final Policy policy = AgentPolicy.getPolicy();
+        if (policy == null) {
+            return; /* noop */
+        }
+
+        String filePath = null;
+        if (args.length > 0 && args[0] instanceof String pathStr) {
+            filePath = Paths.get(pathStr).toAbsolutePath().toString();
+        } else if (args.length > 0 && args[0] instanceof Path path) {
+            filePath = path.toAbsolutePath().toString();
+        }
+
+        if (filePath == null) {
+            return; // No valid file path found
+        }
+
+        final StackWalker walker = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
+        final List<ProtectionDomain> callers = walker.walk(new StackCallerChainExtractor());
+
+        boolean isMutating = method.getName().startsWith("write")
```

Comment:
Create a constant Set with all of these values then simplify this to `boolean isMutating = actionSet.contains(method.getName())`